### ISSUE TITLE
Counting parameters

### ISSUE THREAD

**Issue by sebastienwood** [NVIDIA/apex#342]
Hi,
 this is not a bug but a feature request ! :) 
since APEX manages by itself the ops precision, it would be great to have access to a summary of the number of bits that the APEXified model requires to run. Like `sum(p.numel() for p in model.parameters() if p.requires_grad)` in Pytorch, but taking into account the quantization that is used.

That would help to qualify on the go the gains in memory for example.

**Comment by mcarilli** [comment]
Unfortunately I don't think Amp has a good way to do this.  The memory high-water mark is usually not determined by the size of the parameter set itself, but rather, by the size of the fully-built graph just before `loss.backward()` (with lots of intermediate activations saved for backward).

I think the easiest way to evaluate GPU memory use is Pytorch's existing [memory management API](https://pytorch.org/docs/stable/notes/cuda.html#memory-management).
Run a few iterations, then call [torch.cuda.max_memory_allocated()](https://pytorch.org/docs/stable/cuda.html#torch.cuda.max_memory_allocated()) to get the high-water mark of memory that was actually used by your network's allocated tensors.  Try this, for example, with `opt_level=O0`, then again with `opt_level=O1`, to compare pure fp32 vs mixed precision.
